lib/beamchmark/suite/measurements/cpu_info.ex

```diff
@@ -79,17 +57,42 @@ defmodule Beamchmark.Suite.Measurements.CpuInfo do
         acc + map.average_all_cores
       end) / length(cpu_usage_unstable_list)
 
+    sum_by_core =
+      Enum.reduce(cpu_usage_unstable_list, %{}, fn %{cpu_usage: cpu_usage}, sum_cores_acc ->
+        cpu_usage |> reduce_cpu_usage(sum_cores_acc)
+      end)
+
+    number_of_snapshots = length(cpu_usage_unstable_list)
+
+    average_by_core =
+      Enum.reduce(sum_by_core, %{}, fn {core_id, value}, acc ->
+        Map.put(acc, core_id, Float.round(value / number_of_snapshots, 2))
+      end)
+
     %__MODULE__{
       cpu_snapshots: cpu_usage_unstable_list,
+      average_by_core: average_by_core,
       average_all: average_all
     }
   end
 
   @spec diff(t(), t()) :: t()
   def diff(base, new) do
+    average_by_core_diff =
+      Enum.reduce(new.average_by_core, %{}, fn {core_id, value}, acc ->
+        Map.put(acc, core_id, value - Map.get(base.average_by_core, core_id))
+      end)
```

Comment:
```suggestion
      Enum.reduce(new.average_by_core, %{}, fn {core_id, new_core_avg}, average_by_core_diff ->
        Map.put(average_by_core_diff, core_id, Math.percent_diff(new_core_avg, Map.fetch!(base.average_by_core, core_id)))
      end)
```      

Try to name accumulators and values in `Enum.reduce` - it's easier to read it.


You can use `Map.fetch!` as an assertion - we know that in `base.average_by_core` there must be something under key `core_id` and if there isn't anything then it's a bug. In such situations we can use functions with `!` to make sure that our state is correct.